### PR TITLE
LibWeb: Fully implement QueuingStrategies for Streams

### PR DESCRIPTION
**Pull request by shannonbooth**
This implements the remaining functionality for QueuingStrategies in Web::Streams

@mattco98  I've lifted out your relevant changes from the branch you sent me on discord, and resolved the conflicts (with small edit using `vm.names.byteLength` instead), and squashed in the tests I added - was a much better implementation than my attempt! Added you as a Co-Author for those commits. Hope that's all okay with you!

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/Window.cpp [hunk 1]
```diff
@@ -110,6 +110,7 @@ void Window::visit_edges(JS::Cell::Visitor& visitor)
         visitor.visit(plugin_object);
     for (auto& mime_type_object : m_pdf_viewer_mime_type_objects)
         visitor.visit(mime_type_object);
+    visitor.visit(m_count_queuing_strategy_size_function);
```
Comment (mattco98) [inline Userland/Libraries/LibWeb/HTML/Window.cpp:113]: Missing a visit to the byte length strategy

**Review by mattco98 — CHANGES_REQUESTED** [review]
LGTM, aside from one small thing

> @mattco98 I've lifted out your relevant changes from the branch you sent me on discord, and resolved the conflicts (with small edit using vm.names.byteLength instead), and squashed in the tests I added - was a much better implementation than my attempt! Added you as a Co-Author for those commits. Hope that's all okay with you!

Of course! Though it looks like I need to add that email to the mailmap 😄

File: Tests/LibWeb/Text/input/Streams/CountQueuingStrategy-enqueue-desired-size.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<script src="../include.js"></script>
```
Comment (kennethmyhra) [inline Tests/LibWeb/Text/input/Streams/CountQueuingStrategy-enqueue-desired-size.html:1]: The test would imo be nice to have split out in a separate commit

File: Tests/LibWeb/Text/input/Streams/ByteLengthQueuingStrategy-enqueue-desired-size.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<script src="../include.js"></script>
```
Comment (kennethmyhra) [inline Tests/LibWeb/Text/input/Streams/ByteLengthQueuingStrategy-enqueue-desired-size.html:1]: Same for this test

**Review by kennethmyhra — CHANGES_REQUESTED** [review]
Some nit on splitting out the tests in two separate commits.

Probably would have also split out the implementations `byte_length_queuing_strategy_size_function()` and `count_queuing_strategy_size_function()` on the `global object`. But not too important

File: Userland/Libraries/LibWeb/HTML/Window.cpp (hunk 1 above)
Comment (shannonbooth) [reply Userland/Libraries/LibWeb/HTML/Window.cpp:113]: whoops!

File: Tests/LibWeb/Text/input/Streams/CountQueuingStrategy-enqueue-desired-size.html (hunk 2 above)
Comment (shannonbooth) [reply Tests/LibWeb/Text/input/Streams/CountQueuingStrategy-enqueue-desired-size.html:1]: done and done!

**Review by mattco98 — APPROVED** [review]

**Review by kennethmyhra — APPROVED** [review]
